src/components/map/MapboxMap.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useContext, useEffect, useRef, useState } from "react";
+import { AppContext } from "context";
+import mapboxgl from "mapbox-gl";
+import { getEsriVectorSourceStyle, addEsriLayersFromVectorSourceStyle } from "utils/EsriMappingUtil";
+import Legend from "components/map/Legend"
+import Countries from "components/map/Layers/Countries"
+import StudyPins from "components/map/Layers/StudyPins"
+import "components/map/MapboxMap.css";
+import "mapbox-gl/dist/mapbox-gl.css";
+
+mapboxgl.accessToken = process.env.REACT_APP_MAPBOX_API_KEY as string;
+
+const WHO_BASEMAP =
+  "https://tiles.arcgis.com/tiles/5T5nSi527N4F7luB/arcgis/rest/services/WHO_Polygon_Basemap_no_labels/VectorTileServer";
+const WHO_COUNTRY_VECTORTILES =
+  "https://tiles.arcgis.com/tiles/5T5nSi527N4F7luB/arcgis/rest/services/Countries/VectorTileServer";
+
+function mapOnLoad(map: mapboxgl.Map) {
+  getEsriVectorSourceStyle(WHO_COUNTRY_VECTORTILES).then((style) => {
+    addEsriLayersFromVectorSourceStyle(style, map);
+    const styleJson: any = map.getStyle();
+    if (styleJson && styleJson.layers) {
+      for (let layer of styleJson.layers) {
+        const t = layer['source-layer'];
+        if (t === "DISPUTED_AREAS") {
+          map.moveLayer('Countries', layer.id); // HACK for now, moves countries layer behind border once loaded.
+          break;
+        }
+      }
+    }
+  });
+}
+
+const MapboxGLMap = (): any => {
+  const mapContainerRef = useRef<HTMLDivElement>(null);
+  const [state] = useContext(AppContext);
+  const [map, setMap] = useState<mapboxgl.Map | undefined>(undefined);
+
+  // Creates map
+  useEffect(() => {
+    (async () => {
+      const baseMapStyle = await getEsriVectorSourceStyle(WHO_BASEMAP);
+
+      const m = new mapboxgl.Map({ 
+        //@ts-ignore
+        container: mapContainerRef.current, 
+        style: baseMapStyle,
+        center: [0, 30],
+        zoom: 1,
+      });
+
+      m.on("load", () => {
+        mapOnLoad(m)
+        setMap(m);
+      });
+    })();
+
+    return () => map?.remove();
+  },[]);
+
+  Countries(map, state.explore.estimateGradePrevalences);
```

Comment:
Hmm instead of checking for the existence of map in each of the useEffect hooks in the Countries/StudyPins components, why don't we only call these functions when map is not undefined? e.g.

```
if(map){
   Countries(...)
   StudyPins(...)
}
```

Then we could expect map to never be undefined in the context of Countries and StudyPins?

Reply:
Resolved per slack discussion:

![image](https://user-images.githubusercontent.com/21212898/109352199-6e64a480-7848-11eb-966d-28986d41cbf0.png)
![image](https://user-images.githubusercontent.com/21212898/109352215-76244900-7848-11eb-8b0f-8a1de1a693b5.png)